client/logic/ui/game_input.gd

```diff
@@ -85,12 +87,17 @@ func _unhandled_input(event: InputEvent) -> void:
 	if not input_enabled:
 		return
 	if button_enabled:
-		if event.is_action_pressed("select_first"):
-			is_just_first = true
-		if event.is_action_pressed("select_second"):
-			is_just_second = true
+		if selection_enabled and selection.is_mouse_still_inside():
+			if event.is_action_pressed("select_first"):
+				is_just_first = true
+				input_first.emit(await selection.last_terrain)
+				input_dragging.emit(await selection.last_terrain)
+			if event.is_action_pressed("select_second"):
+				is_just_second = true
+				input_second.emit(await selection.last_terrain)
 		if event.is_action_pressed("escape"):
 			is_just_escape = true
+			input_escape.emit()
```

Comment:
more tan input_escape, won't be better input_back ?

Reply:
Escape brings also up the menu

Reply:
ok. I think will be nice to write down the diferent interactions in some document maybe, for the new people that enters the project?

Reply:
That's what https://github.com/Codename-Recon/Codename-Recon/wiki/Game-Structure#map-structure is for.

Reply:
well, this was not map-structure, but it's fine.